tests/conftest.py

```diff
@@ -387,6 +388,60 @@ def shutdown(self):
                                 + " and had to be killed") from None
 
 
+class HcIdle:
+    def __init__(self, hlwm):
+        self.hlwm = hlwm
+        self.proc = subprocess.Popen([hlwm.HC_PATH, '--idle'],
+                                     stdout=subprocess.PIPE,
+                                     bufsize=0)
+        # we don't know how fast self.proc connects to hlwm.
+        # So we keep sending messages via hlwm util we receive some
+        number_sent = 0
+        while [] == select.select([self.proc.stdout], [], [], 0.1)[0]:
+            # while there hasn't been a message received, send something
+            number_sent += 1
+            self.hlwm.call(['emit_hook', 'hc_idle_bootup', number_sent])
+        # now that we know that self.proc is connected, we need to consume
+        # its output up to the last message we have sent
+        assert number_sent > 0
+        number_received = 0
+        while number_received < number_sent:
```

Comment:
So it's guaranteed that all emitted hooks are echoed by the process eventually? Isn't that the same assumption that proved to be wrong in the first place (i.e., an expected hook being emitted before `hc --idle` was even ready)?

Going by you explanation of the race condition, I would have thought that, for that exact same reason, you could *not* wait for `number_received == number_sent` here.